syncer/syncer.go

```diff
@@ -1169,6 +1238,38 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		utils.OsExit(1)
 	})
 
+	updateTSOffset := func() error {
+		t1 := time.Now()
+		ts, tsErr := s.fromDB.getServerUnixTS(ctx)
+		rtt := time.Since(t1).Seconds()
+		if tsErr == nil {
+			s.tsOffset.Store(time.Now().Unix() - ts - int64(rtt/2))
+		}
+		return tsErr
+	}
+	// before sync run, we get the tsoffset from upstream first
+	if utErr := updateTSOffset(); utErr != nil {
+		return utErr
+	}
+	// start background task to get/update current ts offset between dm and upstream
```

Comment:
Can we move this background thread outside syncDML? Since there are multiple syncDML threads and we only need check it once

Reply:
@GMHDBJD `updateTSOffset` is in `syncer.Run` not in `syncDML` 😳 and i think there is only one goroutine running updateTSOffset

Reply:
GitHub review UI is misleading 😂

Reply:
Yeah, I mean L1255-L1272